### PR TITLE
[IA-3327] Added Nextflow to GATK

### PR DESCRIPTION
**Pull request by cstrandw**
Removed Nextflow installation from AoU image and put it into the GATK image

Tested by building and running a container locally and going through the Nextflow tutorial (https://www.nextflow.io/docs/latest/getstarted.html) to verify that Nextflow runs as expected. Saw Nextflow Hello World work correctly.

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]
🎉

**Review by calbach — COMMENTED** [review]
AoU change looks good to me. What's the criteria for bumping the minor version rather than the patch? Feels like a relatively minor change to me (especially relatively to other changes I've seen here, which only bump the patch number).

**Comment by cstrandw** [comment]
> AoU change looks good to me. What's the criteria for bumping the minor version rather than the patch? Feels like a relatively minor change to me (especially relative to other changes I've seen here, which only bump the patch number).

We usually use patch version bumps for bug fixes. Because we are moving where we install Nextflow, we decided to go with a minor bump for both images affected since their tools (adding and removing Nextflow) changed.

**Comment by Qi77Qi** [comment]
I don't feel strongly..but this PR doesn't really remove any package, it just adds a library...so patch would've been fine too

**Review by calbach — APPROVED** [review]
OK. Could we do 2.0.15 for AoU instead? From our image's perspective, this change is effectively a no-op.

**Comment by cstrandw** [comment]
> OK. Could we do 2.0.15 for AoU instead? From our image's perspective, this change is effectively a no-op.

Yes, just pushed that change. Thanks!

**Comment by calbach** [comment]
Per slack, doesn't look related to your changes, but looks like a legitimate issue that would likely result in a failure build the docker image if we merged.

```
W: GPG error: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A4B469963BF863CC
```

This is not a repo that we added ourselves, it was added in a base image. It's curious that we're not seeing issues building those images, perhaps it's due to a particular package we're installing, or perhaps we are and they just aren't being tested here.

**Comment by cstrandw** [comment]
> Per slack, doesn't look related to your changes, but looks like a legitimate issue that would likely result in a failure build the docker image if we merged.
> 
> ```
> W: GPG error: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A4B469963BF863CC
> ```
> 
> This is not a repo that we added ourselves, it was added in a base image. It's curious that we're not seeing issues building those images, perhaps it's due to a particular package we're installing, or perhaps we are and they just aren't being tested here.

Got it, that makes sense. Do you have any suggestions about how we should proceed with this?

File: terra-jupyter-r/Dockerfile
Comment (Qi77Qi) [inline terra-jupyter-r/Dockerfile:124]: we're updating r version in this PR too?

Comment (gpcarr) [reply terra-jupyter-r/Dockerfile:124]: bioconductor released 3.15 which got pulled in automatically to `terra-jupyter-r` - this line just checks the version of bioconductor that gets installed

Comment (Qi77Qi) [reply terra-jupyter-r/Dockerfile:124]: why are we updating bioconductor in this PR?

Comment (gpcarr) [reply terra-jupyter-r/Dockerfile:124]: we aren't explicitly updating the version - without any changes to the r image, [this line ](https://github.com/DataBiosphere/terra-docker/blob/master/terra-jupyter-r/Dockerfile#L124)fails because we're installing bioconductor version 3.15

alternatively, we could try to pin version 3.14 in [this line](https://github.com/DataBiosphere/terra-docker/blob/master/terra-jupyter-r/Dockerfile#L122) to avoid bumping the bioconductor version

Comment (Qi77Qi) [reply terra-jupyter-r/Dockerfile:124]: ah yea...pinning that version is probably better... @nturaga has a seaparate PR for uprading bioconductor

Comment (gpcarr) [reply terra-jupyter-r/Dockerfile:124]: @nturaga's [PR](https://github.com/DataBiosphere/terra-docker/pull/318) only updates Bioconductor in the Anvil RStudio image and not in the terra-jupyter-r image

Comment (Qi77Qi) [reply terra-jupyter-r/Dockerfile:124]: he hasn't PRed that one yet...I paired with him the other day

Comment (nturaga) [reply terra-jupyter-r/Dockerfile:124]: Yes, this is because the R-4.2.0 update on the apt repository isn't available yet. We need to wait to do this with `apt`. The other way to do this is to do a source installation of R, which would involve compiling everything from scratch.

Comment (Qi77Qi) [reply terra-jupyter-r/Dockerfile:124]: let's separate things...and pin the version in https://github.com/DataBiosphere/terra-docker/blob/master/terra-jupyter-r/Dockerfile#L122 for this one

Comment (gpcarr) [reply terra-jupyter-r/Dockerfile:124]: OH that makes SO much sense. I updated bioconductor for terra-jupyter-r and that was causing issues so I'll go ahead and try to pin version 3.14 until we're ready to bump R

**Comment by gpcarr** [comment]
> Hi @calbach @yonghaoy, the AoU image is now failing due to an R and Bioconductor version update breaking the Conda installation. The image build is [failing on this line](https://github.com/DataBiosphere/terra-docker/blob/master/terra-jupyter-aou/Dockerfile#L198) with this [error message](https://gist.github.com/cstrandw/3b320ab3ab2877ef402d24426e7d2f8a), because the [installation of Conda on this line](https://github.com/DataBiosphere/terra-docker/blob/master/terra-jupyter-aou/Dockerfile#L194) causes the R version to be reverted to 4.1 when version 4.2 is expected by the new version of bioconductor (3.15).
> 
> @gpcarr spent a couple of days trying to fix this issue through Conda and was unable to find a good solution at this time, so we believe a potential path forward would be to revert all of the changes to the AoU image and create a follow up ticket to resolve this in the future since we do not support Conda. Potential fixes would be fixing the Conda installation or reverting bioconductor to 3.14. Does this sound like a good plan to you, or do you have any other potential solutions to this issue?

@calbach just fyi, we ended up not including the bioconductor and r version bump in this PR so this comment will no longer be relevant here but will come up in the [PR to bump bioconductor and r](https://github.com/DataBiosphere/terra-docker/pull/319) which we are looking to merge after this PR gets merged.

**Comment by gpcarr** [comment]
bioconductor test failed - https://github.com/DataBiosphere/terra-docker/runs/6262730955?check_suite_focus=true

looks like we are trying to install a package called `RMySQL` which requires that we install `libmysql` in the docker image (likely the r image) which we are not currently installing

will leave it up to the bioconductor team if they'd prefer removing the test or adding that package @nturaga maybe you can include that in your [PR](https://github.com/DataBiosphere/terra-docker/pull/319)

@cstrandw we'll be good to merge once aou and gatk-ovtf tests pass!

File: terra-jupyter-base/Dockerfile
Comment (gpcarr) [inline terra-jupyter-base/Dockerfile:101]: This was needed in every python image starting with the base. MarkupSafe released a new version that is not compatible with a package that `samtools` installs. We reinstall samtools or update packages in every image which was why we needed to add this line to all python images.

File: terra-jupyter-r/Dockerfile
Comment (gpcarr) [inline terra-jupyter-r/Dockerfile:42]: we had to pin the R version to 4.1 until bioconductor is ready to release bioconductor 3.15 with r 4.2

File: terra-jupyter-aou/Dockerfile
Comment (Qi77Qi) [inline terra-jupyter-aou/Dockerfile:1]: this is a minor version bump...is it intended?

Comment (gpcarr) [reply terra-jupyter-aou/Dockerfile:1]: yes the minor bump is for gatk only. we bumped aou as a patch https://github.com/DataBiosphere/terra-docker/pull/316/files#diff-111700937e15194bfcc732196ff2c66042b8496f9077dc25383577a79e174092R1

we were always going to bump gatk as a minor bump because nextflow is a big/involved tool that we are now offering in gatk which we did not previously offer in that image

**Pull request merged**
